interpreter/cling/include/cling/Interpreter/Value.h

```diff
@@ -227,50 +192,51 @@ namespace cling {
     /// \brief Determine whether the Value is set and not void.
     //
     /// Determine whether the Value is set and not void.
-    /// Only in this case can getAs() or simplisticCastAs() be called.
+    /// Only in this case can we can get the represented value.
     bool hasValue() const { return isValid() && !isVoid(); }
 
-    /// \brief Get a reference to the value without type checking.
-    /// T *must* correspond to type. Else use simplisticCastAs()!
-    template <typename T>
-    T& getAs() { return getAs((T*)0); }
-
-    /// \brief Get the value without type checking.
-    /// T *must* correspond to type. Else use simplisticCastAs()!
-    template <typename T>
-    T getAs() const { return const_cast<Value*>(this)->getAs<T>(); }
-
-    template <typename T>
-    T*& getAs(T**) { return (T*&)getAs((void**)0); }
-    void*& getAs(void**) { return m_Storage.m_Ptr; }
-    double& getAs(double*) { return m_Storage.m_Double; }
-    long double& getAs(long double*) { return m_Storage.m_LongDouble; }
-    float& getAs(float*) { return m_Storage.m_Float; }
-    long long& getAs(long long*) { return m_Storage.m_LL; }
-    unsigned long long& getAs(unsigned long long*) { return m_Storage.m_ULL; }
-
-    void*& getPtr() { return m_Storage.m_Ptr; }
-    double& getDouble() { return m_Storage.m_Double; }
-    long double& getLongDouble() { return m_Storage.m_LongDouble; }
-    float& getFloat() { return m_Storage.m_Float; }
-    long long& getLL() { return m_Storage.m_LL; }
-    unsigned long long& getULL() { return m_Storage.m_ULL; }
-
+    // FIXME: If the cling::Value is destroyed and it handed out an address that
+    // might be accessing invalid memory.
+    void** getPtrAddress() { return &m_Storage.m_Ptr; }
```

Comment:
```suggestion
    void** getPtrAddress() { return m_Storage.m_Ptr; }
```
This routine seems to be use to replace `getAs<void*>` which returned the value of the pointer field not it's address.  ie. old code:
```
   template <typename T>
    T& getAs() { return getAs((T*)0); }
...
    void*& getAs(void**) { return m_Storage.m_Ptr; }
```

Reply:
Ha, ok, I thought we return a reference to the pointer, eg, the thing we can modify from the outside. The suggested code change does not seem to compile.

Reply:
Yes, it return a reference so:
```
void*& getPtrAddress() { return m_Storage.m_Ptr; }
```
and since the 'write-to' usage have been replace by a setter, I would have thought it would have worked.  How does it fail?

Reply:
I tried that and many tests started to fail locally, I have not dug into this but I suspect that now in CallFunc when the object is managed by us, we call `getPtrAddress` (in `vp_ary.push_back(fArgVals[i].getPtrAddress());`) which then is supposed to be a `void**` array element. Maybe the conversion fail there. Currently we have two uses of `getPtrAddress`. The `fArgVals[i].getPtrAddress()` one does not have a straight-forward replacement and the other one can be replaced. I am really unhappy with that fact but I cannot think of a better way. 

That said, this does not regress in comparison to the master because in the master we handed out a lot more addresses to various elements. I have a fixme note there and I will try to find a solution but I don't think this should block the current PR.

Reply:
Part of my challenge in understanding this change is that I do not 'see' clearly what code is being replaced by the `getPtrAddress`.  I would have expected calls to `getAs<void*>` or similar.

Reply:
I am not sure either. It seems that where we use the non-const version of getAs used to return the location of the underlying storage. I think that was used to set or convert the value when a call is made. getPtrAddress makes this more explicit. Maybe we can work on further improving this when we are more comfortable in understanding what goes on exactly. The llvm13 upgrade offers debuggable jitted code which will help nailing it down. However, I don’t think we should wait for it to be merged before merging this PR.

Reply:
We should open an issue to track this confusion (since it requires more work to understand and we don't want it to hold the merge).